motors/tacho_motor_class.c

```diff
@@ -721,16 +861,17 @@ static ssize_t speed_sp_store(struct device *dev, struct device_attribute *attr,
 			      const char *buf, size_t size)
 {
 	struct tacho_motor_device *tm = to_tacho_motor(dev);
-	int err, speed;
-
-	if (!(BIT(TM_SPEED_REGULATION_ON) & tm->ops->get_speed_regulations(tm->context)))
-		return -EOPNOTSUPP;
+	int err, speed, max_speed;
 
 	err = kstrtoint(buf, 10, &speed);
 	if (err < 0)
 		return err;
 
-	if (speed > 2000 || speed < -2000)
+	err = tm->ops->get_max_speed(tm->context, &max_speed);
+	if (err < 0)
+		return err;
+
+	if (abs(speed) > max_speed)
```

Comment:
Just did the experiment - the medium motor goes to 1400-1450 RPM when connected to a 9V adapter and the LiIon battery. The large motor is about 950